Data/DataModel/Puzzle.cs

```diff
@@ -13,9 +13,34 @@ public class Puzzle
         public bool IsPuzzle { get; set; } = false;
         public bool IsMetaPuzzle { get; set; } = false;
         public bool IsFinalPuzzle { get; set; } = false;
-        public int FirstSolveValue { get; set; } = 0;
-        public int MinValue { get; set; } = 0;
-        public int PerSolvePenalty { get; set; } = 0;
+        public int SolveValue { get; set; } = 0;
+
+        /// <summary>
+        /// Reward if solved: Sometimes displayed publicly, sometimes used internally by meta engine
+        /// </summary>
+        public string Token { get; set; }
+
+        /// <summary>
+        /// Puzzlehunt: module
+        /// Puzzleday: pregame
+        /// </summary>
+        public string Group { get; set; }
```

Comment:
I'd love to have this be an enum of some kind so it can be a drop down in the UI (and avoid string comparisons), maybe pulled from the database. I'll look into what it takes to get a list more strongly typed here (while still having it vary event to event).

Reply:
My comment must be unclear - for Puzzlehunt the Group would be the *name of the module*. So it can't be an enum, though it could be a link to a Groups table if we want to make one. I felt that was overkill though.

Reply:
Updated comment to clarify.

Reply:
Oh, no, it was clear. I'm wondering if we can pull the list of module names from the db or something similar so that every author doesn't have to put in their module name by hand for every puzzle. And I like things to be strongly typed, but thinking about it more I think filling a collection of strings from the db is plenty to avoid the problems I want to avoid.